test/compiler/immutablearray.jl

```diff
@@ -0,0 +1,482 @@
+using Test
+import Core: ImmutableArray, ImmutableVector
+import Core.Compiler: arrayfreeze_tfunc, mutating_arrayfreeze_tfunc, arraythaw_tfunc
+
+@test arrayfreeze_tfunc(Vector{Int}) === ImmutableVector{Int}
```

Comment:
We also want to add more tests that asserts runtime semantics of those ImmutableArray operations.
E.g. `@test_throws ArgumentError Core.arrayfreeze([1,2,3], nothing)`,  `@test_throws TypeError Core.arraythaw([1,2,3])` etc.